homeassistant/components/upnp/binary_sensor.py

```diff
@@ -46,9 +46,9 @@ def __init__(
     @property
     def available(self) -> bool:
         """Return if entity is available."""
-        return super().available and self.coordinator.data.get(WANSTATUS)
+        return super().available and self.coordinator.data[WAN_STATUS]
```

Comment:
Does this always return a `bool`, or can it return `None` as well? https://github.com/home-assistant/core/pull/54780/files#diff-69a3b80725c7ea0312356fe5c51a7c7c421dea38a1d767099579022063a24835R171 seems to return `None` in the else-part.

Reply:
If None is returned then it will end up being False ( True and None -> False)

Reply:
```
~ ❯❯❯ python3 -i
Python 3.8.10 (default, Jun  2 2021, 10:49:15) 
[GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> True and None
>>> False and None
False
```